File: programs/arrays-objects_card-tostring-6.cpp
string toString() const {
  vector<string> ranks = {"invalid", "Ace",  "2",     "3",   "4",
                                    "5",       "6",    "7",     "8",   "9",
                                    "10",      "Jack", "Queen", "King"};
  vector<string> suits = {"Clubs", "Diamonds", "Hearts", "Spades"};
  return ranks[rank] + " of " + suits[suit];
}

File: programs/arrays-objects_card-tostring-6.java
public String toString() {
    String[] ranks = {null, "Ace", "2", "3", "4", "5", "6",
               "7", "8", "9", "10", "Jack", "Queen", "King"};
    String[] suits = {"Clubs", "Diamonds", "Hearts", "Spades"};
    String s = ranks[this.rank] + " of " + suits[this.suit];
    return s;
}